pkg/storage/mysql/mysqlstore.go

```diff
@@ -55,28 +54,86 @@ const (
 	useDBQuery                = "USE "
 )
 
-// NewProvider instantiates Provider
-func NewProvider(dbPath string, opts ...Option) (*Provider, error) {
+// NewProvider instantiates Provider.
+// Example DB Path root:my-secret-pw@tcp(127.0.0.1:3306)/.
+func NewProvider(dbPath string, opts ...Option) (p *Provider, err error) {
```

Comment:
Why use named returns and returning `prov` at the end? This is misleading.

Reply:
@Baha-sk return params were named in order to wrap any error that might have been thrown with another error that may be thrown while attempting to close the db

Reply:
@Baha-sk I've modified the code to `return p, err` - please check